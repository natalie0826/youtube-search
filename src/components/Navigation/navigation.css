* {
    font-family: Geneva, Arial, Helvetica, sans-serif;
}

.main {
    display: flex;
}

.navigation {
    flex-basis: 20%;
    min-width: 300px;
    padding: 40px;
}

.search-block,
.settings-block {
    margin-bottom: 50px;
}

.title-separator {
    display: flex;
    align-items: center;
    text-align: center;
    font-weight: 300;
    font-size: 20px;
    margin-bottom: 20px;
    color: #555;
    text-transform: uppercase;
}

.title-separator::before,
.title-separator::after {
    content: '';
    flex: 1;
    border-bottom: 2px solid #4B99AD;
}

.title-separator::before {
    margin-right: 10px;
}

.title-separator::after {
    margin-left: 10px;
}

.save {
    width: 100%;
    background: #4B99AD;
    padding: 8px 15px 8px 15px;
    margin: 20px 0;
    border: none;
    color: #fff;
}

.info {
    width: 100%;
    margin: 10px 0;
}

.subtitle-content {
    display: inline-block;
    position: relative;
    padding-right: 10px;
}

.subtitle-content:after {
    content: '';
    position: absolute;
    top: 10px;
    width: 300%;
    max-width: 180px;
    margin-left: 10px;
    height: 1px;
    border-bottom: 1px dotted black;
}

.subtitle-info {
    display: inline-block;
    float: right;
}
