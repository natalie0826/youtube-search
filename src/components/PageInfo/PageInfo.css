.row {
    flex-direction: row;
    display: flex;
    margin: 5px 0;
}

.dots {
    flex: 1;
    border-top-style: dotted;
    border-top-width: 1px;
    max-height: 1px;
    margin-top: 0.5em;
}

.dots-center {
    flex: 1;
    flex-direction: row;
    display: flex;
    margin: 0 10px;
}
