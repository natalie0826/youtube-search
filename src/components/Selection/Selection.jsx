import React from 'react';
import PropTypes from 'prop-types';

import './Selection.css';

export const Selection = (props) => {
    Selection.propTypes = {
        items: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.string),
            PropTypes.arrayOf(PropTypes.number)
        ]).isRequired,
        active: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number
        ]).isRequired,
        onItemChanged: PropTypes.func.isRequired,
        title: PropTypes.string.isRequired
    };

    return (
        <div className="selection">
            <span htmlFor="selection">{props.title}</span>
            <select value={props.active} onChange={props.onItemChanged} onBlur={props.onItemChanged}>
                {props.items.map(item => (
                    <option value={item} key={item}>
                        {item}
                    </option>
                ))}
            </select>
        </div>
    );
};
