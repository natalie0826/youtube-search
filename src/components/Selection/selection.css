.selection {
    display: flex;
    flex-direction: column;
    margin: 20px 0;
}
  
.selection label {
    font-size: 20px;
    margin-bottom: 10px;
    color: #555;
}

.selection select {
    width: 100%;
    box-sizing: border-box;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    border:1px solid #BEBEBE;
    padding: 7px;
    margin:0px;
    transition: all 0.30s ease-in-out;
    -webkit-transition: all 0.30s ease-in-out;
    -moz-transition: all 0.30s ease-in-out;
    -ms-transition: all 0.30s ease-in-out;
    -o-transition: all 0.30s ease-in-out;
    outline: none;
}

.selection select:focus {
    -moz-box-shadow: 0 0 8px #88D5E9;
    -webkit-box-shadow: 0 0 8px #88D5E9;
    box-shadow: 0 0 8px #88D5E9;
    border: 1px solid #88D5E9;
    transition: all 0.30s ease-in-out;
    -webkit-transition: all 0.30s ease-in-out;
    -moz-transition: all 0.30s ease-in-out;
    -ms-transition: all 0.30s ease-in-out;
    -o-transition: all 0.30s ease-in-out;
}
