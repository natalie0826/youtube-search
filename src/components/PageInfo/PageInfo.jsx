import React from 'react';
import PropTypes from 'prop-types';

import './PageInfo.css';

export const PageInfo = (props) => {
    PageInfo.propTypes = {
        subtitle: PropTypes.string.isRequired,
        value: PropTypes.number.isRequired,
    };

    return (
        <div className="info">
            <span className="subtitle-content">{props.subtitle}</span>
            <span className="subtitle-info">{props.value}</span>              
        </div>
    )
}
