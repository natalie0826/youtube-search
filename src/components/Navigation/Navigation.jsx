import React from 'react';
import PropTypes from 'prop-types';
import { DebounceInput } from 'react-debounce-input';

import { Selection } from '../Selection/Selection';
import { PageInfo } from '../PageInfo/PageInfo';
import {
    MIN_QUERY_LENGTH,
    MILLISECONDS_TO_WAIT,
    VIDEO_TYPES,
    PER_PAGE_VALUES
} from '../../constants/app';
import './Navigation.css';

export const Navigation = (props) => {
    Navigation.propTypes = {
        totalCount: PropTypes.number.isRequired,
        activeType: PropTypes.string.isRequired,
        searchQuery: PropTypes.string.isRequired,
        perPage: PropTypes.number.isRequired,
        setVideoType: PropTypes.func.isRequired,
        setPerPageValue: PropTypes.func.isRequired,
        updateSearchQueryAndFetch: PropTypes.func.isRequired
    };

    const setVideoType = e => {
        props.setVideoType(e.target.value);
    };

    const setPerPageValue = e => {
        props.setPerPageValue(e.target.value);
    };

    const updateSearchAndFetch = e => {
        updateSearchQueryAndFetch(e.target.value, props.activeType, props.perPage);
    };

    const {
        totalCount,
        searchQuery,
        updateSearchQueryAndFetch,
        activeType,
        perPage
    } = props;

    return (
        <section className="navigation">
            <div className="search-block">
                <div className="title-separator">Start searching</div>
                <DebounceInput
                    placeholder="Search..."
                    className="search"
                    minLength={MIN_QUERY_LENGTH}
                    value={searchQuery}
                    debounceTimeout={MILLISECONDS_TO_WAIT}
                    onChange={updateSearchAndFetch}
                />
            </div>
            <div className="settings-block">
                <div className="title-separator">Settings</div>
                <Selection
                    title="Video type"
                    items={VIDEO_TYPES}
                    active={activeType}
                    onItemChanged={setVideoType}
                />
                <Selection
                    title="Per page"
                    items={PER_PAGE_VALUES}
                    active={perPage}
                    onItemChanged={setPerPageValue}
                />
            </div>
            <div className="page-info-block">
                <div className="title-separator">Page info</div>
                <div className="info">
                    <PageInfo subtitle="Total count" value={totalCount} />
                </div>
            </div>
        </section>
    );
}
