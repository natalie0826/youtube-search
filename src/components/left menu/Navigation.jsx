import React from 'react';
import PropTypes from 'prop-types';

import { Search } from './Search';
import { Selection } from './Selection';
import '../../styles/navigation.css';

export const Navigation = (props) => {
    Navigation.propTypes = {
        searchUpdate: PropTypes.func.isRequired,
        videoTypes: PropTypes.array.isRequired,
        videoTypeChanged: PropTypes.func.isRequired,
        updateSettings: PropTypes.func.isRequired
    };

    return (
        <div className="navigation">
            <h2>Welcome to the video app!</h2>
            <Search searchUpdate={props.searchUpdate} />
            <Selection title="Video type" items={props.videoTypes} onItemChanged={props.videoTypeChanged} />
            <button className="save" onClick={props.updateSettings}>Update settings</button>
        </div>
    );
}
