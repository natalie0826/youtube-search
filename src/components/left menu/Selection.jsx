import React from 'react';
import PropTypes from 'prop-types';

import '../../styles/selection.css';

export const Selection = (props) => {
    Selection.propTypes = {
        items: PropTypes.array.isRequired,
        onItemChanged: PropTypes.func.isRequired,
        title: PropTypes.string.isRequired
    };

    return (
        <div className="selection">
            <label>{props.title}</label>
            <select value="" onChange={props.onItemChanged}>
                {props.items.map(item => (
                    <option value={item} key={item}>{item}</option>
                ))}
            </select>
        </div>
    )
}
