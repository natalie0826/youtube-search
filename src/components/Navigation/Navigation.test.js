import React from 'react';
import sinon from 'sinon';
import { mount, shallow } from 'enzyme';
import { DebounceInput } from 'react-debounce-input';

import { Selection } from '../Selection/Selection';
import { PageInfo } from '../PageInfo/PageInfo';
import Navigation from './Navigation';

describe('Navigation functional component', () => {
    const navigationProps = {
        pageInfo: {
            perPage: 10,
            totalCount: 1000
        },
        videoTypes: ['one', 'two', 'three'],
        activeType: 'one',
        searchQuery: 'smth',
        perPageValues: [8, 12, 16, 20],
        perPage: 12,
        fetchVideos: jest.fn(),
        updateSearchQueryAndFetch: jest.fn(),
        updateSettings: jest.fn()
    };

    let component;

    beforeEach(() => {
        component = shallow(<Navigation {...navigationProps} />);
    });

    it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('renders Navigation component', () => {
        expect(component).toHaveLength(1);
        expect(component.is('section')).toBe(true);
    });

    it('contains Selection, DebounceInput and PageInfo stateless components', () => {
        component = mount(<Navigation {...navigationProps} />);
        expect(component.contains(Selection)).toBe(true);
        expect(component.contains(DebounceInput)).toBe(true);
        expect(component.contains(PageInfo)).toBe(true);
    });

    it('calls updateSettings method after clicking the button', () => {
        const component = shallow(<Navigation {...navigationProps} />);
        const updateSettings = sinon.spy();
        component.setProps({ updateSettings: updateSettings });
        expect(updateSettings.called).toBeFalsy();
        component.find('.save').simulate('click');
        expect(updateSettings.called).toBeTruthy();
    });
});
