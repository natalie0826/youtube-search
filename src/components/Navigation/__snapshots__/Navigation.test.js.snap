// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation functional component matches its snapshot 1`] = `
<section
  className="navigation"
>
  <div
    className="search-block"
  >
    <div
      className="title-separator"
    >
      Start searching
    </div>
    <DebounceInput
      className="search"
      debounceTimeout={500}
      element="input"
      forceNotifyByEnter={true}
      forceNotifyOnBlur={true}
      minLength={3}
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value="smth"
    />
  </div>
  <div
    className="settings-block"
  >
    <div
      className="title-separator"
    >
      Settings
    </div>
    <Component
      active="one"
      items={
        Array [
          "one",
          "two",
          "three",
        ]
      }
      onItemChanged={[Function]}
      title="Video type"
    />
    <Component
      active={12}
      items={
        Array [
          8,
          12,
          16,
          20,
        ]
      }
      onItemChanged={[Function]}
      title="Per page"
    />
    <button
      className="save"
      onClick={[Function]}
    >
      Update settings
    </button>
  </div>
  <div
    className="page-info-block"
  >
    <div
      className="title-separator"
    >
      Page info
    </div>
    <div
      className="info"
    >
      <Component
        subtitle="Per page"
        value={10}
      />
      <Component
        subtitle="Total count"
        value={1000}
      />
    </div>
  </div>
</section>
`;
